app/frontend/wallet/blockchain-explorer.ts

```diff
@@ -487,10 +488,20 @@ const blockchainExplorer = (ADALITE_CONFIG) => {
   }
 
   function getPoolRecommendation(
-    poolHash: string,
-    stakeAmount: Lovelace
+    stakeAddress: Address,
+    stakeAmount: Lovelace,
+    currentDelegatedPoolHash?: string
   ): Promise<PoolRecommendationResponse> {
-    const url = `${ADALITE_CONFIG.ADALITE_BLOCKCHAIN_EXPLORER_URL}/api/account/poolRecommendation/poolHash/${poolHash}/stake/${stakeAmount}`
+    const searchParams = new URLSearchParams({
+      stakeAddress: bechAddressToHex(stakeAddress),
```

Comment:
Will this ever be undefined? BE defines it as optional, and I wonder if it'd happen

Reply:
not really, I just made the BE endpoint as graceful as possible